### DISCUSSION TITLE
How to set initial_chunk_size_bytes for CUDAExecutionProvider?

### DISCUSSION THREAD

**Discussion by smallbraineng** [pykeio/ort#324] [Q&A] [answered]
In the ONNX docs here [https://onnxruntime.ai/docs/get-started/with-c.html](url) there is a setting, `initial_chunk_size_bytes`, that configures the BFC Arena allocator.

There doesn't seem to be an ort way to set this, and `with_arbitrary_config` fails on `CUDAExectionProvider` with `Parse Unknown provider option: \"initial_growth_chunk_size_bytes\".\n"`

What is the right way to do it?

**Comment by decahedron1** [comment, marked as answer]
`ort` currently doesn't have a safe `ArenaCfg` wrapper, though even if it did it seems like `OrtCUDAProviderOptionsV2` doesn't support taking an `ArenaCfg`; only the legacy version does. The only option here might be to use the [unsafe API directly](https://docs.rs/ort-sys/latest/ort_sys/).

**Reply by smallbraineng** [reply]
Appreciate the prompt answer.

unsafe it is, it is time to live a little
